### ISSUE TITLE
Error detected while processing function iced#nrepl#auto#winenter

### ISSUE THREAD

**Issue by devth** [liquidz/vim-iced#117]
Sometimes I need to restart my repl. If I move my cursor to clojure files while I'm restarting my repl I sometimes get this obscure error:

![image](https://user-images.githubusercontent.com/86107/55242436-d8124900-5202-11e9-97e0-051a469d2d0c.png)

```
Timed out.                                                                                                                                                                                                                                                                       
Error detected while processing function iced#nrepl#auto#winenter[7]..iced#nrepl#ns#in[1]..iced#nrepl#ns#name[6]..<SNR>224_ns_name_by_var:
```


Not sure if it's related but I almost always see an error when `:IcedReconnect`ing after restarting my `iced repl`:

```
Timed out.                                                                                                                                                                                                                                                                       
Disconnected.                                                                                                                                                                                                                                                                    
Error detected while processing function iced#nrepl#reconnect[8]..iced#nrepl#connect[25]..339:                                                                                                                                                                                   
line    7: 
connection failed: connection refused  
```

If I run it again it usually succeeds almost immediately.

**Comment by liquidz** [comment]
@devth Thank you for your reporting!

The cause is same here.
https://github.com/liquidz/vim-iced/pull/86#issuecomment-457020395

If you restart your repl, neovim's channel is disconnected, but vim-iced does not have a chance to detect it.
So vim-iced tries to evaluate code, and timed out.

I'll research if there is a way to detect channel disconnection on neovim...

**Comment by liquidz** [comment]
@devth Maybe I found it!
I oversighted `EOF` description in neovim document.
https://neovim.io/doc/user/channel.html

I tried to fix it, so could you try `bugfix/neovim-detect-disconnection` branch?

**Comment by devth** [comment]
@liquidz I tried out the branch and everything looks good! No more delays or error messages when switching focus after killing the `iced repl`. 👍

**Comment by liquidz** [comment]
@devth Thanks!
I'll merge the branch, but it may be late because I can not take time.. :baby:

**Comment by devth** [comment]
Is `neovim-detect-disconnection` ready to merge?

**Comment by liquidz** [comment]
@devth Yes, I'm ready. But because I'm on parental leave, I may not provide rapid support even if I release next version.
https://github.com/liquidz/vim-iced/pull/120

So I wonder if I should release it...

**Comment by devth** [comment]
Enjoy your leave and little one! For what it’s worth I’ve been using this branch since you created it without any problems.

**Comment by liquidz** [comment]
Closed by #120
